### ISSUE TITLE
init_label_function for classification online train

### ISSUE THREAD

**Issue by dongfangduoshou123** [visionml/pytracking#17]
Hi,
In init_label_function, I understand  " (self.pos ) / (self.target_scale * self.img_support_sz)" will normalize the coordinate of target's center to [0,1], But I cannot understand " (self.pos - self.pos.round()) / (self.target_scale * self.img_support_sz)",  what self.pos - self.pos.round() is mean?  Need your explain，Thank you very much!

`
    def init_label_function(self, train_x):
        # Allocate label function
        self.y = TensorList([x.new_zeros(self.params.sample_memory_size, 1, x.shape[2], x.shape[3]) for x in train_x])

        # Output sigma factor
        output_sigma_factor = self.fparams.attribute('output_sigma_factor')
        self.sigma = (self.feature_sz / self.img_support_sz * self.base_target_sz).prod().sqrt() * output_sigma_factor * torch.ones(2)

        # Center pos in normalized coords
        target_center_norm = (self.pos - self.pos.round()) / (self.target_scale * self.img_support_sz)

        # Generate label functions
        for y, sig, sz, ksz, x in zip(self.y, self.sigma, self.feature_sz, self.kernel_size, train_x):
            center_pos = sz * target_center_norm + 0.5 * torch.Tensor([(ksz[0] + 1) % 2, (ksz[1] + 1) % 2])
            for i, T in enumerate(self.transforms[:x.shape[0]]):
                sample_center = center_pos + torch.Tensor(T.shift) / self.img_support_sz * sz
                y[i, 0, ...] = dcf.label_function_spatial(sz, sig, sample_center)

        # Return only the ones to use for initial training
        return TensorList([y[:x.shape[0], ...] for y, x in zip(self.y, train_x)])
`

**Comment by dongfangduoshou123** [comment]
Is the code "self.pos - self.pos.round()" means to transfer the original point from left-up to self.pos.round() ?

**Comment by dongfangduoshou123** [comment]
In code "center_pos = sz * target_center_norm + 0.5 * torch.Tensor([(ksz[0] + 1) % 2, (ksz[1] + 1) % 2])"

"sz * target_center_norm" part transfer the normalized center pos  to feature map coordinate,  what about another part " + 0.5 * torch.Tensor([(ksz[0] + 1) % 2, (ksz[1] + 1) % 2])"  's mean?

**Comment by martin-danelljan** [comment]
Hi.
Just pushed an update that should clarify things a little. There was a small bug that could lead to 0.5 pixel misalignment.

`self.pos.round()` is the center pixel coordinate of the extracted sample. So  `self.pos - self.pos.round()` should correct from this sub-pixel displacement.

The term `0.5 * torch.Tensor([(ksz[0] + 1) % 2` is to correctly handle even and odd kernels. It leads to small misalignments if you remove that, but the IoU-Net largely compensates for it.
